modules/fluentd_pre/main.tf

```diff
@@ -0,0 +1,27 @@
+provider "template" {
+  # See: https://github.com/terraform-providers/terraform-provider-template/blob/v2.0.0/CHANGELOG.md#200-january-14-2019
+  # Need to pin the minimum version for templates/fluent.conf
+  version = "~> 2.0"
+}
+
+data "template_file" "fluentd_conf" {
+  template = file("${path.module}/templates/fluent.conf")
+
+  vars = {
+    elasticsearch_host = var.elasticsearch_host
+    elasticsearch_port = var.elasticsearch_port
+
+    fluentd_port  = 4224
+    fluentd_match = var.fluentd_match
+
+    s3_bucket     = aws_s3_bucket.logs[0].id
```

Comment:
Currently `fluent.conf` actually doesn't provision the possibility of disabling the S3 logs, so setting `logs_s3_enabled` to `0` would be problematic.

Would prefer to keep this optional approach, and I have just realized we could do this:
https://www.terraform.io/docs/configuration/expressions.html#directives
(slightly related: https://www.terraform.io/docs/configuration/functions/templatefile.html)

We can just pass in `var.logs_s3_enabled` to use as a flag to indicate if we want the S3 block in the `fluent.conf` settings.

Reply:
Edited and tested on staging that if `var.log_s3_enabled` is `false`, the corresponding `td-agent.conf` doesn't have the S3 log storage block.